pyproject.toml

```diff
@@ -1,11 +1,19 @@
 [build-system]
 requires = [
+  #"platform_machine=='x86_64'",
   "setuptools>=42",
   "wheel",
-  "numpy",
+  # lowest NumPy we can use for a given Python,
+  # except for more exotic platform (Mac Arm flavors)
+  "numpy==1.20.0; python_version=='3.8' and platform_machine=='x86_64'",
+  "numpy==1.20.0; python_version=='3.9' and platform_machine=='x86_64'",
+  # As per https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
+  # safest to build at 1.21.6 for all platforms
+  "numpy==1.21.6; python_version=='3.10' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
+  "numpy==1.23.2; python_version=='3.11' and platform_machine=='x86_64' and platform_python_implementation != 'PyPy'",
```

Comment:
So I _think_ this will limit things to x86_64 but in a rather annoying way. I.e. it just won't pick up numpy if you try to `pip install` distopia on anything but x86_64**

That might not actually be the behaviour you want here, as it will just prevent anyone from building unless they edit this (i.e. if someone were to try to build it on arm64 knowing that it was likely broken, they probably would face some issues unless they added an entry here).

Personally I generally am not against this level of hand holding.

** at least I _think_, I can't remember if setup.py gets defaulted to when there are missing build requirements.

Reply:
I'm okay with that for now, as long as there is some discouraging mechanism. It likely won't build not on x86_64 Perhaps I will make EXTRA clear in the docs.